libpod/healthcheck.go

```diff
@@ -402,6 +407,10 @@ func (c *Container) updateHealthCheckLog(hcl define.HealthCheckLog, inStartPerio
 				healthCheck.Status = define.HealthCheckUnhealthy
 			}
 		}
+		if hcResult == define.HealthCheckContainerStopped {
+			healthCheck.Status = define.HealthCheckStopped
+			healthCheck.FailingStreak = oldFailingStreak
+		}
```

Comment:
This looks like the wrong ordering, there is no need to reset FailingStreak when you don't update it in the first place.
```
		if hcResult == define.HealthCheckContainerStopped {
			healthCheck.Status = define.HealthCheckStopped
		} else if !inStartPeriod {
			// increment failing streak
			healthCheck.FailingStreak++
			// if failing streak > retries, then status to unhealthy
			if healthCheck.FailingStreak >= c.HealthCheckConfig().Retries {
				healthCheck.Status = define.HealthCheckUnhealthy
			}
		}
		
```